### PR TITLE
feat: add text color detection and replacement tool for PDFs #6139

### LINKED ISSUE #6139: Feature Request: Detect and replace text colors in PDF
### Summary

Add a tool that:
1. **Analyzes** a PDF and lists all text colors used (as hex values, with a preview swatch and usage count)
2. **Allows the user** to select one or more of those colors and replace them with a target color of their choice

### Use Case

A common workflow in print/design environments: a PDF arrives with text in a color that is wrong or non-standard (e.g., a red annotation, a bright blue heading, a Pantone-like spot color rendered as RGB). The user needs to:
- See at a glance which non-black colors are present in the document text
- Replace a specific color (e.g., all red text → dark cyan, or all `#FF0000` → `#000000`) without touching any other content

This is currently not possible in Stirling-PDF. Workarounds require external tools (PyMuPDF scripting, Acrobat Pro, etc.).

### Proposed UI

1. Upload PDF
2. Tool scans text spans and returns a list: `Color #FF0000 — 42 occurrences`, `Color #0000FF — 7 occurrences`, …
3. User selects source color + target color (color picker)
4. Tool outputs the modified PDF

### Technical Notes

- Libraries like **PyMuPDF** (`fitz`) or **Apache PDFBox** can extract per-span color information from the content stream
- Replacement can be done by drawing a white rectangle over the original span and re-inserting the text in the new color (PyMuPDF approach), or by directly rewriting the color operators in the content stream (PDFBox / iText approach)
- Scope: **text only** — no need to touch images, vector graphics, or background fills

### Priority / Complexity

Medium complexity. The color detection pass is straightforward; the replacement requires careful span-level processing to preserve font, size, and position.

Thank you for the great project!

### PR DESCRIPTION
**Pull request by enjoyandlove**
### Summary
This PR introduces a new PDF text color utility that allows users to detect and replace text colors in uploaded documents.
### What’s included
- Added a text color scan step that analyzes PDF text spans and returns:
  - Hex color value (e.g. `#FF0000`)
  - Preview swatch
  - Usage count
- Added a color replacement flow that lets users:
  - Select one or more source text colors
  - Pick a target color via color picker
  - Export a modified PDF with updated text colors
- Scoped to text content only (no images, vector fills, or background graphics)
### Why
Design/print workflows often require fast correction of non-standard text colors (e.g., red annotations, incorrect heading colors).  
Previously this required external tools or scripts. This feature enables that workflow directly in Stirling-PDF.
### Implementation Notes
- Color detection is performed at text-span level.
- Replacement preserves text position/font/size as closely as possible while updating color.
- Processing targets color operators for text content only.
- Non-text objects are intentionally untouched.
### Limitations / Out of Scope
- No replacement for image colors, vector graphics, or page background fills.
- Complex PDFs with unusual font encoding/rendering may have edge cases; behavior is best-effort with layout preservation.
### Test Plan
- [ ] Upload sample PDF with multiple text colors; verify detected color list and counts
- [ ] Replace a single color (e.g. `#FF0000` -> `#000000`) and verify only matching text changes
- [ ] Replace multiple source colors to one target color
- [ ] Verify black text remains unchanged when not selected
- [ ] Verify output PDF opens in common viewers (browser, Acrobat, etc.)
- [ ] Regression test unrelated tools and general PDF processing flow
### Related
- Closes #6139

### CODE REVIEW

**Comment by enjoyandlove** [comment]
Hi, @jbrunton96 !
I’ve just made the necessary fixes to my PR. When you have a moment, could you kindly review it? I’d also be happy to hear if you have any other suggestions or a different approach in mind.
Thank you.

File: app/core/src/main/java/stirling/software/SPDF/controller/api/misc/ReplaceAndInvertColorController.java [hunk 1]
```diff
@@ -65,4 +72,40 @@ public ResponseEntity<StreamingResponseBody> replaceAndInvertColor(
...
+    }
+
+    @AutoJobPostMapping(
+            consumes = MediaType.MULTIPART_FORM_DATA_VALUE,
+            value = "/replace-text-colors")
+    @Operation(
+            summary = "Replace specific PDF text colours",
+            description =
+                    "Replaces selected source text colours with a single target colour while leaving non-text content unchanged. Input:PDF Output:PDF Type:SISO")
+    public ResponseEntity<StreamingResponseBody> replaceTextColors(
```
Comment (Frooodle) [inline app/core/src/main/java/stirling/software/SPDF/controller/api/misc/ReplaceAndInvertColorController.java]: Sorry recently moved from StreamingResponseBody to Resource instead please check the other class's

**Review by Frooodle — CHANGES_REQUESTED** [review]

**Comment by enjoyandlove** [comment]
Hi, @Frooodle 
I have just addressed the requested changes in this PR. 
Could you please review it again? 
Best regards
